pkg/drivers/vfkit/vfkit.go

```diff
@@ -313,7 +313,7 @@ func isBootpdError(err error) bool {
 }
 
 func (d *Driver) Stop() error {
-	if err := d.SetVFKitState("HardStop"); err != nil {
+	if err := d.SetVFKitState("Stop"); err != nil {
```

Comment:
Might be worth to try `HardStop` if  `Stop` fails?

Reply:
Possible, but usually if `Stop` fails vfkit is likely not running, so `HadrStop` will also fail, and it typically fails.

Reply:
I agree we should do a HardStop if this fails, we have similar code 
we have a package called "retry" I think after a two times of retry with normal stop it should do a hard stop

Reply:
here is an example of using it https://github.com/medyagh/minikube/blob/d510d46696d6e3588921d51d8daf41297e7fdc8d/pkg/minikube/machine/stop.go#L72

Reply:
The current code does not do retries in the driver, assuming that the caller doing the retries. Lets fix the issues first in the driver, and then we can evaluate if stop should retry in the driver, and apply it to more drivers.